open_spiel/games/bridge/bridge.h

```diff
@@ -221,14 +208,31 @@ class BridgeGame : public Game {
   std::unique_ptr<State> NewInitialState() const override {
     return std::unique_ptr<State>(
         new BridgeState(shared_from_this(), UseDoubleDummyResult(),
-                        IsDealerVulnerable(), IsNonDealerVulnerable()));
+                        IsDealerVulnerable(), IsNonDealerVulnerable(), NumTricks()));
   }
   int NumPlayers() const override { return kNumPlayers; }
   double MinUtility() const override { return -kMaxScore; }
   double MaxUtility() const override { return kMaxScore; }
   absl::optional<double> UtilitySum() const override { return 0; }
+
+  int GetObservationTensorSize(int num_tricks) const {
+    int kPlayTensorSize =
+        kNumBidLevels                           // What the contract is
+        + kNumDenominations                     // What trumps are
+        + kNumOtherCalls                        // Undoubled / doubled / redoubled
+        + kNumPlayers                           // Who declarer is
+        + kNumVulnerabilities                   // Vulnerability of the declaring side
+        + kNumCards                             // Our remaining cards
+        + kNumCards                             // Dummy's remaining cards
+        + num_tricks * kNumPlayers * kNumCards  // Number of played tricks
+        + kNumTricks                            // Number of tricks we have won
+        + kNumTricks;                           // Number of tricks they have won
+    int kObservationTensorSize = kNumObservationTypes + std::max(kPlayTensorSize, kAuctionTensorSize);
+    return kObservationTensorSize;
```

Comment:
nit: You can probably collapse this into a single line; I would also move the entire function inside `ObservationTensorShape` since it isn't used anywhere else.